bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/uicases/trades/MyTradesScreen.kt

```diff
@@ -0,0 +1,92 @@
+package network.bisq.mobile.presentation.ui.uicases.trades
+
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.navigation.NavHostController
+import bisqapps.shared.presentation.generated.resources.Res
+import bisqapps.shared.presentation.generated.resources.img_no_trades
+import kotlinx.coroutines.flow.StateFlow
+import network.bisq.mobile.domain.data.model.BisqOffer
+import network.bisq.mobile.presentation.ViewPresenter
+import network.bisq.mobile.presentation.ui.components.atoms.BisqButton
+import network.bisq.mobile.presentation.ui.components.atoms.BisqText
+import network.bisq.mobile.presentation.ui.components.layout.BisqScrollLayout
+import network.bisq.mobile.presentation.ui.components.molecules.OfferCard
+import network.bisq.mobile.presentation.ui.theme.BisqTheme
+import org.jetbrains.compose.resources.painterResource
+import org.koin.compose.koinInject
+import org.koin.core.parameter.parametersOf
+import org.koin.core.qualifier.named
+
+interface IMyTrades: ViewPresenter {
+    val myTrades: StateFlow<List<BisqOffer>>
+
+    fun navigateToCurrencyList()
+}
+
+@Composable
+fun MyTradesScreen() {
+    val navController: NavHostController = koinInject(named("RootNavController"))
+    val tabController: NavHostController = koinInject(named("TabNavController"))
+
+    val presenter: IMyTrades = koinInject { parametersOf(navController, tabController)  }
+
+    val myTrades: List<BisqOffer> = presenter.myTrades.collectAsState().value
+
+    LaunchedEffect(Unit) {
```

Comment:
Let's avoid LaunchedEffect in favour of our new composable `RememberPresenterLifecycle` -> I'll update this PR so we can merge it now

Reply:
Similar to creating custom hooks in React - love it!
Thanks for showing this approach. 😊